pkg/sql/contention/event_store.go

```diff
@@ -348,6 +353,27 @@ func (s *eventStore) resolutionIntervalWithJitter() time.Duration {
 	return jitteredInterval
 }
 
+func logResolvedEvents(ctx context.Context, events []contentionpb.ExtendedContentionEvent) {
+	eventsAggregated := make(map[string]time.Duration)
+	for i := range events {
+		event := events[i]
+		// Create a composite key string
+		key := fmt.Sprintf("%d:%d:%d:%s",
```

Comment:
I think you can use a struct as your map key here instead of serializing into a string, then you can avoid the serialization/deserialization cost. all the args here are comparable so if you put them into a 4 field struct, you can use it as the key directly.